app/models/piece.rb

```diff
@@ -42,8 +42,15 @@ def blocked?(x, y)
     end
 
     piece_path.each do |square|
-      return true if game.is_occupied?(square[0], square[1])
+      return true if game.square_occupied?(square[0], square[1])
     end
     false
   end
+
+  def move_to!(x, y) # move and capture
+    @game = game
```

Comment:
Just realized you probably already know about guard clauses since you used one in `move_to!` already...  Ignore me if all that ^ is redundant.  😛